frontend/components/PearlDeposit/SelectPaymentMethod/SelectPaymentMethod.tsx

```diff
@@ -22,15 +32,79 @@ import { TransferCryptoOn } from './TransferCryptoOn';
 const { Title, Text, Paragraph } = Typography;
 
 const SelectPaymentMethodCard = styled(CardFlex)`
-  width: 360px;
+  width: 323px;
   border-color: ${COLOR.WHITE};
   .ant-card-body {
     height: 100%;
   }
 `;
 
-const ShowAmountsToDeposit = () => {
-  const { amountsToDeposit } = usePearlWallet();
+const ShowAmountsToDeposit = ({
+  isOnRamping = false,
+  onRampChainId,
+  onLoadingChange,
+  onFiatAmountChange,
+}: {
+  isOnRamping?: boolean;
+  onRampChainId?: EvmChainId;
+  onLoadingChange?: (isLoading: boolean) => void;
+  onFiatAmountChange?: (amount: number | null) => void;
+}) => {
+  const { amountsToDeposit, walletChainId } = usePearlWallet();
+  const { updateUsdAmountToPay, updateEthAmountToPay } = useOnRampContext();
+  const chainId = onRampChainId || walletChainId;
+
+  // Calculate total native token from amountsToDeposit
+  // Note: Transak only buys native token (ETH).
+  const totalNativeToken = useMemo(() => {
+    if (!chainId || !amountsToDeposit) return 0;
+
+    const chainDetails = asEvmChainDetails(asMiddlewareChain(chainId));
+    const nativeTokenSymbol = chainDetails.symbol as TokenSymbol;
+
+    const nativeTokenAmount = amountsToDeposit[nativeTokenSymbol]?.amount || 0;
```

Comment:
It is already TokenSymbol 🥲 

```suggestion
    const nativeTokenAmount = amountsToDeposit[chainDetails.symbol]?.amount || 0;
```